books/kestrel/c/syntax/validation-information.lisp

```diff
@@ -700,6 +700,101 @@
                                  ((:e tau-system)))))
   :hooks (:fix))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define type-compatiblep ((x typep) (y typep))
+  :returns (yes/no booleanp)
+  :short "Check that two @(see type)s are compatible [C17:6.2.7]."
+  :long
+  (xdoc::topstring
+   (xdoc::p
+    "Informally, two types are compatible if they represent \"the same\"
+     type.")
+   (xdoc::p
+    "Because we currently only model an approximation of C types, our notion of
+     compatibility is also approximate.
+     Specifically, this relation overapproximates true type compatibility.
+     Compatible types should always be recognized as such,
+     but incompatible types may also be recognized.")
+   (xdoc::p
+    "In particular:"
+    (xdoc::ul
+     (xdoc::li "all composite types of a particular kind are currently
```

Comment:
Why "composite"?

Reply:
Perhaps the wrong word. I meant to refer to the non-atomic types: aggregate types, pointers, unions, function types. Compatibility of these types is approximate because the representation of these types is approximate.

I'll clarify that language.

Reply:
Understood. There must an official term (perhaps consisting of a few categories of types) to describe those types you were referring to (some predicates for that are defined in the validation-information.lisp file).

In the standard, 'composite type' has a specific meaning [C17:6.2.7].

Reply:
I see, yes, definitely the wrong word then. I update the wording to specifically mention the types which are approximated.